File: orange_hrm_ts/tests/steps/employee.steps.ts
import { Page, expect } from '@playwright/test';
import { EmployeePage } from '../pages/EmployeePage';
import { waitForHeading } from '../helpers/wait.helper';

export async function criarFuncionario(page: Page) {
  const employee = new EmployeePage(page);
  const firstName = `User${Date.now()}`;
  const lastName = 'Test';

  await employee.goToPIM();
  await employee.clickAddEmployee();
  await employee.fillFirstName(firstName);
  await employee.fillLastName(lastName);
  await employee.save();
  await employee.waitUntilLoaded();

  return `${firstName}`;
}

export async function validarCamposObrigatorios(page: Page) {
  const employee = new EmployeePage(page);

  await employee.goToPIM();
  await employee.clickAddEmployee();
  await employee.save();

  await expect(page.locator('.oxd-input-field-error-message').first()).toBeVisible();
}

File: orange_hrm_PW_Python/tests/steps/employee_steps.py
from playwright.sync_api import expect
from tests.pages.employee_page import EmployeePage
import time

def criar_funcionario(page):
    employee = EmployeePage(page)

    first_name = f"User{int(time.time()*1000)}"
    last_name = "Test"

    employee.go_to_pim()
    employee.click_add_employee()

    employee.fill_employee(first_name, last_name)

    employee.save()

    employee.wait_loaded()

    assert "viewPersonalDetails" in page.url

    return first_name